lib/rtcpeerconnection/firefox.js

```diff
@@ -209,29 +123,50 @@ Object.defineProperty(FirefoxRTCPeerConnection.prototype, 'peerIdentity', {
   })
 });
 
-// NOTE(mmalavalli): Firefox does not support RTCPeerConnection#removeStream(),
-// and calling it will screw up the RTCPeerConnection's internal state. So
-// for now, we don't do anything when it is called.
-// Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=842455
-//
-// UPDATE(mmalavalli): We tried to implement removeStream() using
-// RTCPeerConnection#removeTrack(), but ran into a Firefox bug in the
-// Firefox <--> Chrome interop case, which is mentioned below.
-// Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1133874
-FirefoxRTCPeerConnection.prototype.removeStream = function removeStream() {};
+// NOTE(mmalavalli): Because we are not delegating to the native
+// RTCPeerConnection#removeTrack(), we have to manually maintain a list of added
+// tracks. So we disable the delegation to the native RTCPeerConnection#addTrack()
+// for now.
+FirefoxRTCPeerConnection.prototype.addTrack = function addTrack() {
+  var args = [].slice.call(arguments);
+  var track = args[0];
+  var sender = this._senders.get(track.id);
+  if (sender && sender.track === track) {
+    throw new Error('Cannot add MediaStreamTrack [' + track.id + ', '
+      + track.kind + ']: RTCPeerConnection already has it');
+  }
```

Comment:
Is Firefox not yet handling this error case for us?

Reply:
@markandrus If we don't check here, then we might get an unwanted exception when we try to add a track for which we have called removeTrack() previously (but which is not actually removed).